buildSrc/src/main/kotlin/org/gradle/plugins/PerformanceTestPlugin.kt

```diff
@@ -132,16 +132,15 @@ class PerformanceTestPlugin : Plugin<Project> {
             configureForAnyPerformanceTestTask(this, performanceSourceSet, prepareSamplesTask, performanceReportTask)
             scenarioList = file("$buildDir/$performanceTestScenarioListFileName")
             scenarioReport = file("$buildDir$performanceTestScenarioReportFileName")
-            buildTypeId = propertyOrNull(buildTypeIdProperty)
-            workerTestTaskName = if (project.hasProperty(workerTestTaskNameProperty)) project.property(workerTestTaskNameProperty).toString()
-                    else "fullPerformanceTest"
-            coordinatorBuildId = propertyOrNull(coordinatorBuildIdProperty)
-            branchName = propertyOrNull(branchnameProperty)
+            buildTypeId = stringPropertyOrNull(buildTypeIdProperty)
+            workerTestTaskName = stringPropertyOrNull(workerTestTaskNameProperty) ?: "fullPerformanceTest"
+            coordinatorBuildId = stringPropertyOrNull(coordinatorBuildIdProperty)
+            branchName = stringPropertyOrNull(branchnameProperty)
             teamCityUrl = teamCityUrlValue
-            teamCityUsername = propertyOrNull(teamCityUserNameProperty)
-            teamCityPassword = propertyOrNull(teamCityPasswordProperty)
+            teamCityUsername = stringPropertyOrNull(teamCityUserNameProperty)
+            teamCityPassword = stringPropertyOrNull(teamCityPasswordProperty)
             afterEvaluate {
-                if (branchName != null && !branchName.isEmpty()) {
+                branchName?.takeIf { it.isNotEmpty() }?.let { branchName ->
```

Comment:
Doesn't this cause a NPE if branchName is null?

Reply:
It doesn't thanks to the safe-navigation operator `?.` combined with the `takeIf` and `let` primitives:

    e?.takeIf(pred) => if (e != null && pred(e)) e else null
    e?.let(f) => if (e != null) f(e) else null

Reply:
Very nice. Thanks.